src/Microsoft.ML.TimeSeries/PredictionEngine.cs

```diff
@@ -0,0 +1,253 @@
+﻿using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// A class that runs the previously trained model (and the preceding transform pipeline) on the
+    /// in-memory data, one example at a time.
+    /// This can also be used with trained pipelines that do not end with a predictor: in this case, the
+    /// 'prediction' will be just the outcome of all the transformations.
+    /// </summary>
+    /// <typeparam name="TSrc">The user-defined type that holds the example.</typeparam>
+    /// <typeparam name="TDst">The user-defined type that holds the prediction.</typeparam>
+    public sealed class PredictionEngine<TSrc, TDst>
```

Comment:
>PredictionEngine [](start = 24, length = 16)

This appears to have the same name, and indeed 90% of the same code, as the existing prediction engine. The same name is bad, because that's a name collision between two publicly facing types in our API (albeit in different namespaces). We should avoid that.

Also I feel compelled to point out that there are ways for classes to share code beyond copy and paste.